website/docs/provider_reference.html.markdown

```diff
@@ -0,0 +1,96 @@
+---
+layout: "google"
+page_title: "google provider reference"
+sidebar_current: "docs-google-provider-reference"
+description: |-
+  The Google provider is used to configure your GCP project, location, and creds
+---
+
+# `google` provider reference
+The `google` provider block is used to configure default values for your GCP
+project and location (`zone` and `region`), and add your credentials.
+
+-> You can avoid using a provider block by using environment variables. Every
+field of the `google` provider is optional. If you want to share configs between
+environments and deploy to different projects, try it out!
+
+## Example Usage
+
+```hcl
+provider "google" {
+  credentials = "${file("account.json")}"
+  project     = "my-project-id"
+  region      = "us-central1"
+  zone        = "us-central1-c"
+}
+```
+
+## Configuration Reference
+
+The following keys can be used to configure the provider.
+
+* `credentials` - (Optional) The path or contents of a file that contains your
+  service account private key in JSON format. You can download your existing
+  [Google Cloud service account file] from the Google Cloud Console, or you can
+  create a new one from the same page.
+
+  Credentials can also be specified using any of the following environment
+  variables (listed in order of precedence):
+
+    * `GOOGLE_CREDENTIALS`
+    * `GOOGLE_CLOUD_KEYFILE_JSON`
+    * `GCLOUD_KEYFILE_JSON`
+
+  The [`GOOGLE_APPLICATION_CREDENTIALS`][adc]
+  environment variable can also contain the path of a file to obtain credentials
+  from.
+
+  If no credentials are specified, the provider will fall back to using the
+  [Google Application Default Credentials][adc].
+  If you are running Terraform from a GCE instance, see [Creating and Enabling
+  Service Accounts for Instances][gce-service-account] for details.
+
+  On your computer, if you have made your identity available as the
+  Application Default Credentials by running [`gcloud auth application-default
+  login`][gcloud adc], the provider will use your identity.
+
+  -> [Service accounts][service accounts] are the recommended way
+  to manage GCP credentials. [GCE metadata] is also acceptable, although it can
+  only be used when running Terraform from within [certain GCP resources](https://cloud.google.com/docs/authentication/production#obtaining_credentials_on_compute_engine_kubernetes_engine_app_engine_flexible_environment_and_cloud_functions).
+  Application Default Credentials, such as those obtained through the `gcloud`
```

Comment:
It's not that Application Default Credentials are not guaranteed, it's that _gcloud_ credentials are not guaranteed, and I want to make sure we aren't confusing the two thing.

Application Default Credentials just means the credentials that GCP APIs can find without you having to explicitly pass them in (https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application). It encompasses GCE service accounts, App Engine, gcloud credentials, and anything that sets the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.

gcloud credentials are ones granted from gcloud specifically. _These_ are the ones not guaranteed to work for all APIs, because the project that grants these credentials needs to explicitly enable the API in question.

I'm not sure whether our users have a concept of ADC vs gcloud credentials, so I think we don't necessarily need to use either term. Just saying that credentials obtained through `gcloud` are not guaranteed to work for all APIs, and so using a service account is recommended is probably sufficient.